docker/Dockerfile

```diff
@@ -6,30 +6,61 @@
 # docs/source/assets/contributing/dockerfile-stages-dependency.png
 
 ARG CUDA_VERSION=12.8.1
+ARG BUILD_BASE_IMAGE=nvidia/cuda:${CUDA_VERSION}-devel-ubuntu20.04
+ARG FINAL_BASE_IMAGE=nvidia/cuda:${CUDA_VERSION}-devel-ubuntu22.04
```

Comment:
See my comment here -- I'm concerned about upgrading to Ubuntu 22.04 due to glibc issues

https://github.com/vllm-project/vllm/pull/18092#issuecomment-2877805116